exporter/exporter_test.go

```diff
@@ -117,6 +118,65 @@ func setupDBKeys(t *testing.T, uri string) error {
 	return nil
 }
 
+func setupDBKeysCluster(t *testing.T, uri string) error {
+	log.Printf("Creating cluster object")
+	cluster := redisc.Cluster{
+		StartupNodes: []string{uri},
+	}
+	log.Printf("Running refresh on cluster object")
+	if err := cluster.Refresh(); err != nil {
+		log.Debugf("Cluster refresh failed: %v", err)
+	}
+
+	log.Printf("Creating redis connection object")
+	conn, err := cluster.Dial()
+	if err != nil {
+		log.Printf("Dial failed: %v", err)
+	}
+
+	c, err := redisc.RetryConn(conn, 10, 100*time.Millisecond)
+	if err != nil {
+		log.Printf("RetryConn failed: %v", err)
+	}
+
+	defer c.Close()
+
+	if _, err := c.Do("SELECT", "0"); err != nil {
```

Comment:
Instead of repeating this all here, put the setupKeys() code in a separate function and call it from both setupDBKeys() and setupDBKeysCluster() and pass the connection object as a parameter.